### PR TITLE
feat/Build header component

### LINKED ISSUE #92: Build header component

### PR DESCRIPTION
**Pull request by sumeetrohra**
close #92

## Types of changes

- [x] Add tags to auto close the corresponding issue.
- [x] I followed branch and commit naming convention.
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] I have read the **CONTRIBUTING** document.
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by sumeetrohra** [comment]
A representational component as of now, will add functionality as we make more routes.
captured

File: client/src/components/Header/index.jsx [hunk 1]
```diff
@@ -0,0 +1,59 @@
+import React from 'react';
+import { withRouter } from 'react-router-dom';
```
Comment (Navneet-Sahota) [inline client/src/components/Header/index.jsx]: Why this?

File: client/src/Routes.jsx [hunk 2]
```diff
@@ -6,13 +6,15 @@ import {
...
 import NoMatch from './views/NoMatch';
+import HomePage from './views/HomePage';
 
 function Routes() {
   return (
     <Switch>
       <Route exact path="/" render={() => <Redirect to="/login" />} />
       <Route path="/login" component={Login} />
       <Route path="/signup" component={Signup} />
+      <Route path="/home" component={HomePage} />
```
Comment (Navneet-Sahota) [inline client/src/Routes.jsx]: Name your `Home` instead of `Homepage` for consistency

File: client/src/components/Header/index.jsx [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+const Header = (props) => {
+  const dropDownOptions = [
+    { key: 1, text: 'Profile', value: 1 },
+    { key: 2, text: 'Add Post', value: 2 },
+    { key: 3, text: 'My Posts', value: 3 },
+    { key: 4, text: 'Logout', value: 4 },
+  ];
+
+  const { history } = props;
```
Comment (Navneet-Sahota) [inline client/src/components/Header/index.jsx]: Do destructuring in function parameter itself.

File: client/src/elements/CenterContainer/index.jsx [hunk 4]
```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Container } from 'semantic-ui-react';
+
+const CenterContainer = (props) => {
+  const { children } = props;
```
Comment (Navneet-Sahota) [inline client/src/elements/CenterContainer/index.jsx]: Do destructuring in function parameter itself.

File: client/src/elements/DropDown/index.jsx [hunk 5]
```diff
@@ -0,0 +1,27 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Dropdown, Image } from 'semantic-ui-react';
+
+const DropDown = (props) => {
+  const { triggerImage, options } = props;
```
Comment (Navneet-Sahota) [inline client/src/elements/DropDown/index.jsx]: Same

File: client/src/elements/HeaderMenu/index.jsx [hunk 6]
```diff
@@ -0,0 +1,30 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Menu } from 'semantic-ui-react';
+
+const HeaderMenu = (props) => {
+  const { children } = props;
```
Comment (Navneet-Sahota) [inline client/src/elements/HeaderMenu/index.jsx]: Same

File: client/src/elements/HeaderMenuRight/index.jsx [hunk 7]
```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Menu } from 'semantic-ui-react';
+
+const HeaderMenuRight = (props) => {
+  const { children } = props;
```
Comment (Navneet-Sahota) [inline client/src/elements/HeaderMenuRight/index.jsx]: Same

File: client/src/elements/ImageElement/index.jsx [hunk 8]
```diff
@@ -0,0 +1,8 @@
+import React from 'react';
+import { Image } from 'semantic-ui-react';
+
+const ImageElement = (props) => (
```
Comment (Navneet-Sahota) [inline client/src/elements/ImageElement/index.jsx]: Do we really need this component? It looks like an unnecessary wrapper.

File: client/src/elements/MenuItem/index.jsx [hunk 9]
```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Menu } from 'semantic-ui-react';
+
+const HeaderMenuRight = (props) => {
```
Comment (Navneet-Sahota) [inline client/src/elements/MenuItem/index.jsx]: I don't think this is `HeaderMenuRight`

Comment (sumeetrohra) [reply client/src/elements/MenuItem/index.jsx]: will make a new issue and make changes

File: client/src/elements/ImageElement/index.jsx (hunk 8 above)
Comment (sumeetrohra) [reply client/src/elements/ImageElement/index.jsx]: this is semantics image wrapper, it works a little differently, so we do need a wrapper.

File: client/src/components/Header/index.jsx (hunk 1 above)
Comment (sumeetrohra) [reply client/src/components/Header/index.jsx]: How will you bring history prop in this component, can't really work with 'Link' on onClick...

**Pull request merged**